24 progress bar

Starwhip opened the pull request:
Add progress bar component to main.

kai-kaldahl reviewed (changes requested):
Label values should not be hardcoded, they should be dynamic. There should be some sort of visual indicator on progress bars that are over-filled. Is there a reason as to why you did not use the built in html progress bar? (https://developer.mozilla.org/en-US/docs/Web/HTML/Element/progress)

Starwhip commented:
> Label values should not be hardcoded, they should be dynamic. There should be some sort of visual indicator on progress bars that are over-filled. Is there a reason as to why you did not use the built in html progress bar? (https://developer.mozilla.org/en-US/docs/Web/HTML/Element/progress)

-Label values can be updated dynamically by changing the String parameter - this would be done by back-end when the items update I imagine. Since this is a generic progress bar element, and not one specifically for the allocations page, I did not have the labels reflect current/max values by default, we may want other text inside.

-I can add a styling for bars over-filled tonight.

-HTML progress bar has no options for inner text - I saw a lot of comments on code sites telling people not to use the explicit progress bar element for bars with inside text. (Like here: https://stackoverflow.com/a/41430146)

Starwhip commented:
Added overfill styling as yellow/black "caution tape" border.

danii412 commented:
Just reviewed again, looks good - sending back to QA

danii412 commented on Plot/Components/PartialComponents/ProgressBar/ProgressBar.razor:
```diff
@@ -0,0 +1,98 @@
+    </div>
+
+    <div class="col-6">
+        <div class="progress-container @_barClass">
```
I really want us to instead of using all of these divs & classes to use the ```meter``` tag. It is perfect for what we are looking for. Here are some resources on using it. Let me know if you have any questions.
https://www.w3schools.com/tags/tryit.asp?filename=tryhtml5_meter
https://stackoverflow.com/questions/43983062/how-to-make-text-in-meter-tag-visible-over-the-meter

Starwhip commented:
Rewrote to use <meter> element. Adjusted bar styling to prevent height changes during element width changes.
Only thing that seems like it could use improvement is to set a text-wrap rule on the label, or adjust column rules (not in the design document, but we could have the bar sit full width under the label), so that the icon doesn't sit on its own line when the text is too long, and the width of the bar is too short.
![image](https://github.com/user-attachments/assets/b604b6a6-b144-49e6-9a79-e3130a5528bb)

claytonleonardcook reviewed (changes requested):
@Starwhip Sorry to pass this back but I think this might be the last thing. This needs to be fixed because the ID shouldn't have to be required for the component to work. I see you want to pass the background color to the meter element so you can style it so you have that style block in there. I also see you're targeting it via the ID on it which isn't great because then it doesn't work without one. So, here's a better solution I think and I would recommend as a method of giving styles via props:

```html
<div class="ProgressBar row @Class" id="@ID" style="--progress-bar-background-color: @barColor;">
...
</div>
```

```css
.ProgressBar {
  --progress-bar-background-color: #f00; // some kind of default, if needed.
}

meter::-moz-meter-bar,
meter::-webkit-meter-even-less-good-value,
meter::-webkit-meter-suboptimum-value,
meter::-webkit-meter-optimum-value {
  background: var(--progress-bar-background-color);
}
```

Starwhip commented:
Should be ready for review
